### ISSUE TITLE
New Logging Module or Wrapper for Logrus to send logs to different streams

### ISSUE THREAD

**Issue by iam-veeramalla** [argoproj/argo-cd#5793]
# Summary
ArgoCD does not send logs to different log streams(stdout and stderr) depending on the log severity. There are couple of open issues on this, example - #3491. 

While working on the PR,  I realised that there is no proper way to achieve this using the Logrus module. There are open issues on the Logrus module but the maintainers does not seems to be much interested to get this fix landed. 
https://github.com/sirupsen/logrus/issues/403

There are couple of workarounds to get this done.
1. Use the hooks provided by Logrus  (https://github.com/sirupsen/logrus/tree/master/hooks/writer).
2. Write a wrapper that reads the LogLevel and writes to appropriate log stream.

I tried to implement both with argo-cd. Below are my findings.
1. There is a downside for Hooks approach. Hooks creates duplication of logs. Every log get printed twice, we can manage to send one of the logs to /dev/null, but it will degrade the performance. 
2. I tried to write a wrapper for the logrus module and the PR is open for review (https://github.com/argoproj/argo-cd/pull/5574). I believe this will also introduce some kind of performance degradation but I haven't tested it with many applications.

What change you think needs making.
Either we go with https://github.com/argoproj/argo-cd/pull/5574  or We look at a new logging module.

# Motivation
#3491 

# Proposal
There is the other side of the coin. Many people don't agree with sending logs to different streams. While working on the issue, I looked at many forums and communities with different opinions. Many believe that all the logs should be logged to stderr.
https://www.gnu.org/software/libc/manual/html_node/Standard-Streams.html
https://pubs.opengroup.org/onlinepubs/9699919799/functions/stderr.html
https://linux.die.net/man/3/stderr

While the other opinion is to have different log streams for applications. Logging tools like kibana add more power to this argument.
https://docs.python.org/3/library/sys.html#sys.stderr
ipfs/go-log#60

I had a discussion with @jannfis regarding this and we thought it's better to open a enhancement request with all the observations and decide if we want to go with 
https://github.com/argoproj/argo-cd/pull/5574

or 

New logging module

**Comment by edoger** [comment]
With the current state of logrus, a new log library should be necessary.

**Comment by iam-veeramalla** [comment]
@edoger true, I was evaluating zap logger. I found it good and also recommended by others in the community. I have started migrating in steps, I can collaborate if you are interested.
